python/src/space/core/ops/change_data.py

```diff
@@ -52,24 +52,26 @@ def read_change_data(
         f"End snapshot ID {end_snapshot_id} should not be lower than start "
         f"snapshot ID {start_snapshot_id}")
 
-  all_snapshot_ids = sorted(storage.snapshot_ids)
+  all_snapshot_ids = []
+  current_snapshot = storage.snapshot(end_snapshot_id)
+  while current_snapshot.snapshot_id >= start_snapshot_id:
+    all_snapshot_ids.insert(0, current_snapshot.snapshot_id)
+    if not current_snapshot.HasField("parent_snapshot_id"):
```

Comment:
Is it better to use _INIT_SNAPSHOT_ID (https://github.com/google/space/blob/main/python/src/space/core/storage.py#L51) than `HasField("parent_snapshot_id")`? We can break if current_snapshot is _INIT_SNAPSHOT_ID (we can change it to INIT_SNAPSHOT_ID to avoid style warning).

I am not sure whether a default parent_snapshot_id = 0 and an explicit parent_snapshot_id = 0 have any difference for HasField("parent_snapshot_id"). It is a little confusing.

Reply:
It will be different when snapshot expiration is implemented. The stype check is mainly complaining about no types given for all_snapshot_ids